server/interpreter.js

```diff
@@ -2213,6 +2227,61 @@ Interpreter.prototype.initThread_ = function() {
       return intrp.createArrayFromList(callers, perms);
     }
   });
+
+  // Properties of the Thread prototype object.
+  /**
+   * Decorator to add standard permission and type checks for Thread
+   * prototype methods.
+   * @param {!Interpreter.NativeCallImpl} func Function to decorate.
+   * @return {!Interpreter.NativeCallImpl} The decorated function.)
+   */
+  var withChecks = function(func) {
+    name = (name === undefined ? func.name : name);
```

Comment:
Elsewhere in the code we typically do:
`  x = (a === b) ? c : d;`
Not that it matters.